src/components/PersonLInk.tsx

```diff
@@ -0,0 +1,79 @@
+import { Link, useParams, useSearchParams } from 'react-router-dom';
+import cn from 'classnames';
+
+import { useMemo } from 'react';
+import { Person } from '../types';
+
+type Props = {
+  person: Person;
+  filteredPeople: Person[];
+};
+
+export const PersonLink: React.FC<Props> = ({ person, filteredPeople }) => {
+  const { born, died, fatherName, motherName, name, sex, slug } = person;
+
+  const { slugParam } = useParams();
+  const [searchParams] = useSearchParams();
+
+  const hasMother = useMemo(
+    () => filteredPeople.find(p => p.name === motherName),
+    [filteredPeople, motherName],
+  );
+
+  const hasFather = useMemo(
+    () => filteredPeople.find(p => p.name === fatherName),
+    [filteredPeople, fatherName],
+  );
+
+  return (
+    <tr
+      data-cy="person"
+      key={name}
+      className={cn({ 'has-background-warning': slug === slugParam })}
+    >
+      <td>
+        <Link
+          className={cn({ 'has-text-danger': sex === 'f' })}
+          to={{
+            pathname: `/people/${slug}`,
+            search: searchParams.toString(),
+          }}
+        >
+          {name}
+        </Link>
+      </td>
+      <td>{sex}</td>
+      <td>{born}</td>
+      <td>{died}</td>
+      <td>
+        {hasMother ? (
+          <Link
+            className={cn({ 'has-text-danger': hasMother.sex === 'f' })}
+            to={{
+              pathname: `/people/${hasMother.slug}`,
```

Comment:
note that one test related to this link failed. Try to fix it

Reply:
1) 
       #/people/:correct-slug page
         should allow to select another person:
     AssertionError: Timed out retrying after 4000ms: Expected to find element: `[data-cy="person"]`, but never found it.
      at Context.eval (webpack://react_people-table-advanced/./cypress/support/commands.ts:39:0)
      
      
      
      
      I have no idea how to fix it. It says that I'm missing the data-cy attribute, but I'm not. All the tests passed well in the previous version of the boilerplate.